src/main/java/com/exadel/ipromise/controllers/ReasonController.java

```diff
@@ -0,0 +1,61 @@
+package com.exadel.ipromise.controllers;
+
+
+import com.exadel.ipromise.dto.ReasonDto;
+import com.exadel.ipromise.exception.FieldValidationPromiseException;
+import com.exadel.ipromise.service.ReasonService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+
+@RestController
+@CrossOrigin(origins = "http://localhost:8080")
+@RequestMapping("/reason")
+public class ReasonController {
+
+    private final ReasonService reasonService;
+
+    public ReasonController(ReasonService reasonService) {
+        this.reasonService = reasonService;
+    }
+
+
+    @PostMapping
+    public ReasonDto create(@Valid @RequestBody ReasonDto reasonDto, BindingResult result) {
+
+        validation(result);
+        return reasonService.addReason(reasonDto);
```

Comment:
i would recommend you use ResponseEntity in all endpoints. It will be more consistent and i think better for client when you will be waiting the same format of response. But it's up to you

Reply:
Done